### ISSUE TITLE
Failed to construct 'HTMLElement' new DomModule

### ISSUE THREAD

**Issue by shobson** [webpack-contrib/polymer-webpack-loader#53]
So I've updated to the latest version 1.2.6 but I now seem to be getting some errors in console. Now I am on windows 7 and it was working before I updated. I ran a npm run build to compile the files and im loading the bundle.js script on an html page and trying to view <my-element> on there as well. The error message below is all im seeing now.

`bundle.js:3075 Uncaught TypeError: Failed to construct 'HTMLElement': Please use the 'new' operator, this DOM object constructor cannot be called as a function.
    at new DomModule (bundle.js:3075)
    at Function.register (bundle.js:109)
    at Object.<anonymous> (bundle.js:916)
    at __webpack_require__ (bundle.js:20)
    at Object.<anonymous> (bundle.js:890)
    at __webpack_require__ (bundle.js:20)
    at bundle.js:63
    at bundle.js:66
DomModule @ bundle.js:3075
register @ bundle.js:109
(anonymous) @ bundle.js:916
__webpack_require__ @ bundle.js:20
(anonymous) @ bundle.js:890
__webpack_require__ @ bundle.js:20
(anonymous) @ bundle.js:63
(anonymous) @ bundle.js:66
bundle.js:6483 Uncaught TypeError: Failed to construct 'HTMLElement': Please use the 'new' operator, this DOM object constructor cannot be called as a function.
    at MyElement.PropertyAccessors (bundle.js:6483)
    at MyElement.TemplateStamp (bundle.js:7393)
    at MyElement.PropertyEffects (bundle.js:4291)
    at MyElement.PolymerElement (bundle.js:2346)
    at new MyElement (bundle.js:926)
    at Object.<anonymous> (bundle.js:951)
    at __webpack_require__ (bundle.js:20)
    at Object.<anonymous> (bundle.js:890)
    at __webpack_require__ (bundle.js:20)
    at bundle.js:63`

**Comment by robotgryphon** [comment]
Having the same issue after upgrading. Using latest Webpack and all NPM packages fully up to date.

**Comment by bryandcoulter** [comment]
Can you guys double check that the es5 adapter and polyfills are loading in the correct order. https://github.com/webpack-contrib/polymer-webpack-loader/blob/master/demo/src/index.ejs

**Comment by shobson** [comment]
I just checked. I never changed that file so it looks like the one you just included pointing to the demo/src/index.ejs file.

**Comment by bryandcoulter** [comment]
hmm, I am having trouble replicating it. Can you send me a sample repo?

**Comment by shobson** [comment]
I should note that I am not seeing this error when running the webpack-dev-server. I am only seeing this error when I trying to initialise <my-element> and loading the script bundle.js file into an existing project framework. This was working before back in 1.2.4 version as I recall. Maybe I am missing something?

**Comment by bryandcoulter** [comment]
The only way I can force the browser to throw that error is if the browser is not loading the custom-elements-es5-adapter.js for some reason.

**Comment by shobson** [comment]
I just manually added the script /bower_components/webcomponentsjs/custom-elements-es5-adapter.js where I am calling the bundle.js script and it seems to have resolved that issue. Is that supposed to be included in the bundle.js? I assume its a dependency?

**Comment by bryandcoulter** [comment]
![screen shot 2017-08-18 at 9 27 32 am](https://user-images.githubusercontent.com/18359726/29463247-ad7cea90-83f7-11e7-90e4-0cb281ca1d80.png)

In this case I altered the path to adapter to ensure it would not load.

**Comment by bryandcoulter** [comment]
You don't want it in the bundle because it is not necessary for basically ie11. we use the 

```js
new CopyWebpackPlugin([{
    from: path.resolve(__dirname, 'bower_components/webcomponentsjs/*.js'),
    to: 'bower_components/webcomponentsjs/[name].[ext]'
 }])
```
and then this section here in the index file to ensure that it only gets loaded on the browsers that needed it. https://github.com/webpack-contrib/polymer-webpack-loader/blob/master/demo/src/index.ejs#L17

**Comment by shobson** [comment]
So whats the best way of resolving this issue? I am using chrome version 60.0.3112.90 and without me manually loading custom-elements-es5-adapter.js in my site along side the bundle.js it will throw those errors. So rather than manually importing the scripts, bundle.js and custom-elements-es5 in my site, should i be just importing the index.html that gets created on build? <link rel="import" href="index.html">

**Comment by bryandcoulter** [comment]
I know it is not pretty but I have not found a better way of doing it.

**Comment by ChadKillingsworth** [comment]
Loading the polyfills is extremely tricky. Here's all the requirements:

- The custom-elements-es5-adapter.js must be included first. You shouldn't bundle this with other scripts because it will actually throw exceptions in Internet Explorer 11 (this is by design).
- Your main script bundle should not be included in the page until the WebComponentsReady event is seen. This indicates that the polyfills have all loaded. The loader is asynchronous, that's why you have to wait.

You will see different results based on timing (which is why you don't see it in the dev server). These issues are more pronounced with the webpack loader because HTML Imports adds a delay for you.

Regardless, these issues are all limitations of the polyfills. I suggest you follow the demo exactly here. There are other ways to do it, but none of them are straight forward because of the restrictions I listed.

This is probably worth listing in the readme though.

**Comment by shobson** [comment]
Okay thank you for the clarification. Just to be sure, in order for me then to get this into another project framework I will have to import index.html and have that file determine the scripts to load?

**Comment by robotgryphon** [comment]
I'm getting the errors while running the dev server. Nothing while compiling the project but a heap when loading it into Chrome. I have a repo up on Bitbucket you can clone if desired.

**Comment by ChadKillingsworth** [comment]
Closed by #55

**Comment by robdodson** [comment]
I hit this failed to construct DomModule issue today and realized it's because I was loading the custom-elements-es5 adapter but I wasn't actually transpiling my code.

If you have an element named `MyElement`, look in your bundled js to see if it still has `class MyElement` or if it has the transpiled version. If it still has `class MyElement` then you may have forgotten to include a `.babelrc`. Here's what I use:

```json
{
  "presets": ["es2015"]
}
```

And you need to also make sure you're chaining your loaders. In your webpack config do:

```
module: {
  rules: [
    {
      test: /\.html$/,
      use: [
        // Chained loaders are applied last to first
        { loader: 'babel-loader' },
        { loader: 'polymer-webpack-loader' }
      ]
    },
  ...
  ]
}
```
